openduck-py/openduck_py/routers/voice.py

```diff
@@ -277,59 +269,76 @@ def _inference(sentence: str):
             messages = chat.history_json["messages"]
             messages.append(new_message)
 
-            response = await acompletion(CHAT_MODEL, messages, temperature=0.3)
-
-            response_message = response.choices[0].message
-            completion = response_message.content
-            t_chat = time()
-            await log_event(
-                db,
-                self.session_id,
-                "generated_completion",
-                meta={"text": completion},
-                latency=t_chat - t_whisper,
-            )
-            print(
-                f"Used {response.usage.prompt_tokens} prompt tokens and {response.usage.completion_tokens} completion tokens"
+            response = await asyncio.to_thread(
+                completion, CHAT_MODEL, messages, temperature=0.3, stream=True
```

Comment:
I think this might be a better approach due to `async` issues https://litellm.vercel.app/docs/providers/ollama#example-usage---streaming--acompletion